acl test failed with users pace permission check

**What happened**:
https://github.com/juicedata/juicefs/actions/runs/12536004628/job/34958531893
```
root@bench-01:~/juicefs# ./juicefs format sqlite3://test.db --enable-acl --trash-days 0 myjfs
2024/12/29 22:21:42.979200 juicefs[1383806] <INFO>: Meta address: sqlite3://test.db [NewClient@interface.go:513]
2024/12/29 22:21:42.979783 juicefs[1383806] <INFO>: Data use file:///var/jfs/myjfs/ [format@format.go:509]
2024/12/29 22:21:43.057224 juicefs[1383806] <INFO>: Volume is formatted as {
  "Name": "myjfs",
  "UUID": "476b96f8-08c5-4c64-a30f-dd661e46e98d",
  "Storage": "file",
  "Bucket": "/var/jfs/",
  "BlockSize": 4096,
  "Compression": "none",
  "EncryptAlgo": "aes256gcm-rsa",
  "TrashDays": 0,
  "MetaVersion": 1,
  "MinClientVersion": "1.2.0-A",
  "DirStats": true,
  "EnableACL": true
} [format@format.go:546]
root@bench-01:~/juicefs# ./juicefs mount -d sqlite3://test.db /tmp/jfs --non-default-permission
2024/12/29 22:22:05.997683 juicefs[1383821] <INFO>: Meta address: sqlite3://test.db [NewClient@interface.go:513]
2024/12/29 22:22:05.999255 juicefs[1383821] <INFO>: Data use file:///var/jfs/myjfs/ [mount@mount.go:639]
2024/12/29 22:22:06.499628 juicefs[1383821] <INFO>: OK, myjfs is ready at /tmp/jfs [checkMountpoint@mount_unix.go:225]
root@bench-01:~/juicefs# python3 .github/scripts/hypo/acl_test.py
setup_logger ./fs1.log
setup_logger ./fs2.log
__init__
duration is 0.04049086570739746
2024-12-29 22:22:14,586 - INFO - do_listdir /tmp/fsrand/ succeed
2024-12-29 22:22:14,587 - INFO - do_listdir /tmp/jfs/fsrand/ succeed
2024-12-29 22:22:14,600 - INFO - do_change_groups user1 root [] succeed
2024-12-29 22:22:14,608 - INFO - do_change_groups user1 root [] succeed
2024-12-29 22:22:14,608 - INFO - run_cmd: sudo -u root setfacl       -m u:root:-,g:root:r,o::- /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-12-29 22:22:15,341 - INFO - do_set_acl /tmp/fsrand/ with u:root:-,g:root:r,o::- succeed
2024-12-29 22:22:15,341 - INFO - run_cmd: sudo -u root setfacl       -m u:root:-,g:root:r,o::- /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-12-29 22:22:16,061 - INFO - do_set_acl /tmp/jfs/fsrand/ with u:root:-,g:root:r,o::- succeed
2024-12-29 22:22:16,061 - INFO - do_listdir /tmp/fsrand/ succeed
2024-12-29 22:22:16,062 - INFO - do_listdir /tmp/jfs/fsrand/ succeed
.__init__
duration is 1.562835693359375
2024-12-29 22:22:16,102 - INFO - do_create_file /tmp/fsrand/aaaa with mode w succeed
2024-12-29 22:22:16,114 - INFO - do_create_file /tmp/jfs/fsrand/aaaa with mode w succeed
2024-12-29 22:22:16,115 - INFO - run_cmd: sudo -u root setfacl  -R     -m u:root:-,g:root:r,o::- /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-12-29 22:22:16,123 - INFO - do_set_acl /tmp/fsrand/ with u:root:-,g:root:r,o::- succeed
2024-12-29 22:22:16,123 - INFO - run_cmd: sudo -u root setfacl  -R     -m u:root:-,g:root:r,o::- /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-12-29 22:22:16,855 - INFO - do_set_acl /tmp/jfs/fsrand/ with u:root:-,g:root:r,o::- succeed
2024-12-29 22:22:16,855 - INFO - run_cmd: sudo -u root setfacl       -m u:root:-,g:user1:r,o::- /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-12-29 22:22:16,863 - INFO - do_set_acl /tmp/fsrand/ with u:root:-,g:user1:r,o::- succeed
2024-12-29 22:22:16,863 - INFO - run_cmd: sudo -u root setfacl       -m u:root:-,g:user1:r,o::- /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-12-29 22:22:16,874 - INFO - do_set_acl /tmp/jfs/fsrand/ with u:root:-,g:user1:r,o::- succeed
2024-12-29 22:22:16,874 - INFO - run_cmd: sudo -u user1 setfacl  -R     -m u:root:-,g:root:r,o::- /tmp/fsrand/
2024-12-29 22:22:16,882 - INFO - do_set_acl /tmp/fsrand/ {'user_perm': '-', 'group_perm': 'r', 'other_perm': '-'} failed:
setfacl: /tmp/fsrand//aaaa: Permission denied
2024-12-29 22:22:16,882 - INFO - run_cmd: sudo -u user1 setfacl  -R     -m u:root:-,g:root:r,o::- /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-12-29 22:22:16,891 - INFO - do_set_acl /tmp/jfs/fsrand/ with u:root:-,g:root:r,o::- succeed
F__init__
duration is 2.389054536819458
2024-12-29 22:22:16,939 - INFO - do_chmod /tmp/fsrand/ 0o0 root succeed
2024-12-29 22:22:16,942 - INFO - do_chmod /tmp/jfs/fsrand/ 0o0 root succeed
2024-12-29 22:22:16,955 - INFO - do_change_groups user1 root [] succeed
2024-12-29 22:22:16,963 - INFO - do_change_groups user1 root [] succeed
2024-12-29 22:22:16,964 - INFO - run_cmd: sudo -u root setfacl       -m u:root:-,g:root:r,o::- /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-12-29 22:22:16,972 - INFO - do_set_acl /tmp/fsrand/ with u:root:-,g:root:r,o::- succeed
2024-12-29 22:22:16,972 - INFO - run_cmd: sudo -u root setfacl       -m u:root:-,g:root:r,o::- /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-12-29 22:22:16,984 - INFO - do_set_acl /tmp/jfs/fsrand/ with u:root:-,g:root:r,o::- succeed
2024-12-29 22:22:16,984 - INFO - do_create_file /tmp/fsrand/aaaa with mode w succeed
2024-12-29 22:22:16,991 - INFO - do_create_file /tmp/jfs/fsrand/aaaa with mode w succeed
2024-12-29 22:22:16,991 - INFO - do_listdir /tmp/fsrand/ succeed
2024-12-29 22:22:16,992 - INFO - do_listdir /tmp/jfs/fsrand/ {'user': 'user1'} failed: [Errno 13] Permission denied: '/tmp/jfs/fsrand/'
F__init__
duration is 2.492546319961548
2024-12-29 22:22:17,053 - INFO - do_create_file /tmp/fsrand/aaaa with mode w succeed
2024-12-29 22:22:17,061 - INFO - do_create_file /tmp/jfs/fsrand/aaaa with mode w succeed
2024-12-29 22:22:17,061 - INFO - run_cmd: sudo -u root setfacl  -R     -m u:user1:wxr,g:root:r,o::wxr,m::- /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-12-29 22:22:17,070 - INFO - do_set_acl /tmp/fsrand/ with u:user1:wxr,g:root:r,o::wxr,m::- succeed
2024-12-29 22:22:17,070 - INFO - run_cmd: sudo -u root setfacl  -R     -m u:user1:wxr,g:root:r,o::wxr,m::- /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-12-29 22:22:17,089 - INFO - do_set_acl /tmp/jfs/fsrand/ with u:user1:wxr,g:root:r,o::wxr,m::- succeed
2024-12-29 22:22:17,089 - INFO - run_cmd: sudo -u user1 setfacl  -R     -m u:root:-,g:root:r,o::- /tmp/fsrand/
2024-12-29 22:22:17,097 - INFO - do_set_acl /tmp/fsrand/ {'user_perm': '-', 'group_perm': 'r', 'other_perm': '-'} failed:
setfacl: /tmp/fsrand//aaaa: Operation not permitted
setfacl: /tmp/fsrand/: Operation not permitted
2024-12-29 22:22:17,097 - INFO - run_cmd: sudo -u user1 setfacl  -R     -m u:root:-,g:root:r,o::- /tmp/jfs/fsrand/
2024-12-29 22:22:17,109 - INFO - do_set_acl /tmp/jfs/fsrand/ {'user_perm': '-', 'group_perm': 'r', 'other_perm': '-'} failed:
setfacl: /tmp/jfs/fsrand//aaaa: Operation not permitted
setfacl: /tmp/jfs/fsrand/: Operation not permitted
.__init__
duration is 2.6071207523345947
2024-12-29 22:22:17,169 - INFO - do_create_file /tmp/fsrand/lbca with mode w succeed
2024-12-29 22:22:17,176 - INFO - do_create_file /tmp/jfs/fsrand/lbca with mode w succeed
2024-12-29 22:22:17,176 - INFO - run_cmd: sudo -u root setfacl  -R --mask --no-mask  -P -m u:root:wxr,g:user1:wr,o::-,m::wxr /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-12-29 22:22:17,185 - INFO - do_set_acl /tmp/fsrand/ with u:root:wxr,g:user1:wr,o::-,m::wxr succeed
2024-12-29 22:22:17,185 - INFO - run_cmd: sudo -u root setfacl  -R --mask --no-mask  -P -m u:root:wxr,g:user1:wr,o::-,m::wxr /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-12-29 22:22:17,201 - INFO - do_set_acl /tmp/jfs/fsrand/ with u:root:wxr,g:user1:wr,o::-,m::wxr succeed
2024-12-29 22:22:17,201 - INFO - do_chmod /tmp/fsrand/ 0o4004 root succeed
2024-12-29 22:22:17,204 - INFO - do_chmod /tmp/jfs/fsrand/ 0o4004 root succeed
2024-12-29 22:22:17,204 - INFO - run_cmd: sudo -u user1 setfacl -d -R    -P -m u:user2:-,g:group4:x,o::-,m::xw /tmp/fsrand/
2024-12-29 22:22:17,212 - INFO - do_set_acl /tmp/fsrand/ {'user_perm': '-', 'group_perm': 'x', 'other_perm': '-'} failed:
setfacl: /tmp/fsrand//lbca: Permission denied
setfacl: /tmp/fsrand/: Operation not permitted
2024-12-29 22:22:17,212 - INFO - run_cmd: sudo -u user1 setfacl -d -R    -P -m u:user2:-,g:group4:x,o::-,m::xw /tmp/jfs/fsrand/
2024-12-29 22:22:17,221 - INFO - do_set_acl /tmp/jfs/fsrand/ {'user_perm': '-', 'group_perm': 'x', 'other_perm': '-'} failed:
setfacl: /tmp/jfs/fsrand/: Operation not permitted
F__init__
duration is 2.71822452545166
2024-12-29 22:22:17,280 - INFO - do_create_file /tmp/fsrand/aaca with mode w succeed
2024-12-29 22:22:17,288 - INFO - do_create_file /tmp/jfs/fsrand/aaca with mode w succeed
2024-12-29 22:22:17,288 - INFO - do_set_xattr /tmp/fsrand/aaca user.0 b'abc' 2 succeed
2024-12-29 22:22:17,291 - INFO - do_set_xattr /tmp/jfs/fsrand/aaca user.0 b'abc' 2 succeed
2024-12-29 22:22:17,291 - INFO - run_cmd: sudo -u root setfacl       -m u:root:r,g:root:r,o::- /tmp/fsrand/aaca
run_cmd:getfacl /tmp/fsrand/aaca
2024-12-29 22:22:17,300 - INFO - do_set_acl /tmp/fsrand/aaca with u:root:r,g:root:r,o::- succeed
2024-12-29 22:22:17,300 - INFO - run_cmd: sudo -u root setfacl       -m u:root:r,g:root:r,o::- /tmp/jfs/fsrand/aaca
run_cmd:getfacl /tmp/jfs/fsrand/aaca
2024-12-29 22:22:17,312 - INFO - do_set_acl /tmp/jfs/fsrand/aaca with u:root:r,g:root:r,o::- succeed
2024-12-29 22:22:17,312 - INFO - run_cmd: sudo -u root setfacl --remove-all /tmp/fsrand/aaca
2024-12-29 22:22:17,319 - INFO - do_remove_acl /tmp/fsrand/aaca with --remove-all succeed
run_cmd:getfacl /tmp/fsrand/aaca
2024-12-29 22:22:17,320 - INFO - run_cmd: sudo -u root setfacl --remove-all /tmp/jfs/fsrand/aaca
2024-12-29 22:22:17,330 - INFO - do_remove_acl /tmp/jfs/fsrand/aaca with --remove-all succeed
run_cmd:getfacl /tmp/jfs/fsrand/aaca
2024-12-29 22:22:17,332 - INFO - do_list_xattr /tmp/fsrand/aaca succeed
2024-12-29 22:22:17,332 - INFO - do_list_xattr /tmp/jfs/fsrand/aaca succeed
.__init__
duration is 2.8283028602600098
2024-12-29 22:22:17,389 - INFO - run_cmd: sudo -u root setfacl -d -R --mask    -m u:user1:wxr,g:root:-,o::-,m::- /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-12-29 22:22:17,397 - INFO - do_set_acl /tmp/fsrand/ with u:user1:wxr,g:root:-,o::-,m::- succeed
2024-12-29 22:22:17,397 - INFO - run_cmd: sudo -u root setfacl -d -R --mask    -m u:user1:wxr,g:root:-,o::-,m::- /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-12-29 22:22:17,410 - INFO - do_set_acl /tmp/jfs/fsrand/ with u:user1:wxr,g:root:-,o::-,m::- succeed
2024-12-29 22:22:17,410 - INFO - do_create_file /tmp/fsrand/afds with mode w succeed
2024-12-29 22:22:17,418 - INFO - do_create_file /tmp/jfs/fsrand/afds with mode w succeed
.__init__
duration is 2.9145359992980957
2024-12-29 22:22:17,476 - INFO - do_create_file /tmp/fsrand/stsn with mode x succeed
2024-12-29 22:22:17,483 - INFO - do_create_file /tmp/jfs/fsrand/stsn with mode x succeed
2024-12-29 22:22:17,483 - INFO - run_cmd: sudo -u root setfacl -d -R --mask    -m u:root:r,g:group4:x,o::-,m::w /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-12-29 22:22:17,491 - INFO - do_set_acl /tmp/fsrand/ with u:root:r,g:group4:x,o::-,m::w succeed
2024-12-29 22:22:17,491 - INFO - run_cmd: sudo -u root setfacl -d -R --mask    -m u:root:r,g:group4:x,o::-,m::w /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-12-29 22:22:17,504 - INFO - do_set_acl /tmp/jfs/fsrand/ with u:root:r,g:group4:x,o::-,m::w succeed
2024-12-29 22:22:17,504 - INFO - do_create_file /tmp/fsrand/qpyt with mode w succeed
2024-12-29 22:22:17,511 - INFO - do_create_file /tmp/jfs/fsrand/qpyt with mode w succeed
2024-12-29 22:22:17,512 - INFO - do_copy_file /tmp/fsrand/stsn /tmp/fsrand/knmh succeed
2024-12-29 22:22:17,522 - INFO - do_copy_file /tmp/jfs/fsrand/stsn /tmp/jfs/fsrand/knmh succeed
2024-12-29 22:22:17,523 - INFO - do_open /tmp/fsrand/qpyt [512] 2579 succeed
2024-12-29 22:22:17,526 - INFO - do_open /tmp/jfs/fsrand/qpyt [512] 2579 succeed
.__init__
duration is 3.0233728885650635
2024-12-29 22:22:17,601 - INFO - run_cmd: sudo -u user1 setfacl -d -R   -L  -m u:user2:wxr,g:root:r,o::xr,m::r /tmp/fsrand/
2024-12-29 22:22:17,609 - INFO - do_set_acl /tmp/fsrand/ {'user_perm': 'wxr', 'group_perm': 'r', 'other_perm': 'xr'} failed:
setfacl: /tmp/fsrand/: Operation not permitted
2024-12-29 22:22:17,609 - INFO - run_cmd: sudo -u user1 setfacl -d -R   -L  -m u:user2:wxr,g:root:r,o::xr,m::r /tmp/jfs/fsrand/
2024-12-29 22:22:17,618 - INFO - do_set_acl /tmp/jfs/fsrand/ {'user_perm': 'wxr', 'group_perm': 'r', 'other_perm': 'xr'} failed:
setfacl: /tmp/jfs/fsrand/: Operation not permitted
2024-12-29 22:22:17,618 - INFO - do_create_file /tmp/fsrand/abha with mode a succeed
2024-12-29 22:22:17,625 - INFO - do_create_file /tmp/jfs/fsrand/abha with mode a succeed
2024-12-29 22:22:17,625 - INFO - run_cmd: sudo -u root setfacl   --mask --no-mask   -m u:user1:-,g:user3:x,o::- /tmp/fsrand/abha
run_cmd:getfacl /tmp/fsrand/abha
2024-12-29 22:22:17,634 - INFO - do_set_acl /tmp/fsrand/abha with u:user1:-,g:user3:x,o::- succeed
2024-12-29 22:22:17,634 - INFO - run_cmd: sudo -u root setfacl   --mask --no-mask   -m u:user1:-,g:user3:x,o::- /tmp/jfs/fsrand/abha
run_cmd:getfacl /tmp/jfs/fsrand/abha
2024-12-29 22:22:17,655 - INFO - do_set_acl /tmp/jfs/fsrand/abha with u:user1:-,g:user3:x,o::- succeed
2024-12-29 22:22:17,656 - INFO - do_list_xattr /tmp/fsrand/abha succeed
2024-12-29 22:22:17,656 - INFO - do_list_xattr /tmp/jfs/fsrand/abha succeed
.__init__
duration is 3.1532797813415527
2024-12-29 22:22:17,713 - INFO - do_chmod /tmp/fsrand/ 0o6333 root succeed
2024-12-29 22:22:17,717 - INFO - do_chmod /tmp/jfs/fsrand/ 0o6333 root succeed
2024-12-29 22:22:17,717 - INFO - run_cmd: sudo -u user1 setfacl --remove-default /tmp/fsrand/
2024-12-29 22:22:17,725 - INFO - do_remove_acl /tmp/fsrand/ with --remove-default succeed
run_cmd:getfacl /tmp/fsrand/
2024-12-29 22:22:17,726 - INFO - run_cmd: sudo -u user1 setfacl --remove-default /tmp/jfs/fsrand/
2024-12-29 22:22:17,733 - INFO - do_remove_acl /tmp/jfs/fsrand/ with --remove-default succeed
run_cmd:getfacl /tmp/jfs/fsrand/
2024-12-29 22:22:17,734 - INFO - do_mkdir /tmp/fsrand/uopt with mode 0o2142 succeed
2024-12-29 22:22:17,738 - INFO - do_mkdir /tmp/jfs/fsrand/uopt with mode 0o2142 succeed
2024-12-29 22:22:17,738 - INFO - do_chown /tmp/fsrand/uopt user1 succeed
2024-12-29 22:22:17,741 - INFO - do_chown /tmp/jfs/fsrand/uopt user1 succeed
2024-12-29 22:22:17,759 - INFO - do_change_groups user1 group4 ['group2'] succeed
2024-12-29 22:22:17,768 - INFO - do_change_groups user1 group4 ['group2'] succeed
2024-12-29 22:22:17,768 - INFO - run_cmd: sudo -u user1 setfacl   --mask --no-mask   -m u::-,g:group2:wxr,o::x /tmp/fsrand/uopt
run_cmd:getfacl /tmp/fsrand/uopt
2024-12-29 22:22:17,776 - INFO - do_set_acl /tmp/fsrand/uopt with u::-,g:group2:wxr,o::x succeed
2024-12-29 22:22:17,776 - INFO - run_cmd: sudo -u user1 setfacl   --mask --no-mask   -m u::-,g:group2:wxr,o::x /tmp/jfs/fsrand/uopt
run_cmd:getfacl /tmp/jfs/fsrand/uopt
2024-12-29 22:22:17,789 - INFO - do_set_acl /tmp/jfs/fsrand/uopt with u::-,g:group2:wxr,o::x succeed
.__init__
duration is 3.294661521911621
2024-12-29 22:22:17,837 - INFO - run_cmd: sudo -u root setfacl -d      -m u::r,g::-,o::- /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-12-29 22:22:17,845 - INFO - do_set_acl /tmp/fsrand/ with u::r,g::-,o::- succeed
2024-12-29 22:22:17,845 - INFO - run_cmd: sudo -u root setfacl -d      -m u::r,g::-,o::- /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-12-29 22:22:17,857 - INFO - do_set_acl /tmp/jfs/fsrand/ with u::r,g::-,o::- succeed
2024-12-29 22:22:17,857 - INFO - do_mkdir /tmp/fsrand/aaaa with mode 0o0 succeed
2024-12-29 22:22:17,861 - INFO - do_mkdir /tmp/jfs/fsrand/aaaa with mode 0o0 succeed
.__init__
duration is 3.3572700023651123
2024-12-29 22:22:17,900 - INFO - do_create_file /tmp/fsrand/aaaa with mode w succeed
2024-12-29 22:22:17,907 - INFO - do_create_file /tmp/jfs/fsrand/aaaa with mode w succeed
2024-12-29 22:22:17,907 - INFO - run_cmd: sudo -u root setfacl       -m u:user1:-,g:root:-,o::- /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-12-29 22:22:17,916 - INFO - do_set_acl /tmp/fsrand/ with u:user1:-,g:root:-,o::- succeed
2024-12-29 22:22:17,916 - INFO - run_cmd: sudo -u root setfacl       -m u:user1:-,g:root:-,o::- /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-12-29 22:22:17,928 - INFO - do_set_acl /tmp/jfs/fsrand/ with u:user1:-,g:root:-,o::- succeed
2024-12-29 22:22:17,928 - INFO - do_chmod /tmp/fsrand/ 0o4 root succeed
2024-12-29 22:22:17,931 - INFO - do_chmod /tmp/jfs/fsrand/ 0o4 root succeed
2024-12-29 22:22:17,932 - INFO - run_cmd: sudo -u user1 setfacl  -R     -m u:root:-,g:root:-,o::- /tmp/fsrand/
2024-12-29 22:22:17,939 - INFO - do_set_acl /tmp/fsrand/ {'user_perm': '-', 'group_perm': '-', 'other_perm': '-'} failed:
setfacl: /tmp/fsrand//aaaa: Permission denied
setfacl: /tmp/fsrand/: Operation not permitted
2024-12-29 22:22:17,939 - INFO - run_cmd: sudo -u user1 setfacl  -R     -m u:root:-,g:root:-,o::- /tmp/jfs/fsrand/
2024-12-29 22:22:17,948 - INFO - do_set_acl /tmp/jfs/fsrand/ {'user_perm': '-', 'group_perm': '-', 'other_perm': '-'} failed:
setfacl: /tmp/jfs/fsrand/: Operation not permitted
F
======================================================================
FAIL: test_acl_1006 (__main__.TestFsrand2.test_acl_1006)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/juicefs/.github/scripts/hypo/acl_test.py", line 32, in test_acl_1006
    state.set_acl(default=False, entry=v1, group='root', group_perm={'r'}, logical=False, mask=set(), not_recalc_mask=False, other_perm=set(), physical=False, recalc_mask=False, recursive=True, set_mask=False, sudo_user='user1', user='root', user_perm=set())
  File "/root/juicefs/.github/scripts/hypo/fs.py", line 417, in set_acl
    target=EntryWithACL,
         ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/dist-packages/hypothesis/stateful.py", line 736, in rule_wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/root/juicefs/.github/scripts/hypo/fs.py", line 417, in set_acl
    target=EntryWithACL,
         ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/dist-packages/hypothesis/stateful.py", line 857, in precondition_wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/root/juicefs/.github/scripts/hypo/fs.py", line 438, in set_acl
    assert self.equal(result1, result2), f'\033[31mset_acl:\nresult1 is {result1}\nresult2 is {result2}\033[0m'
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: set_acl:
result1 is
setfacl: /tmp/fsrand//aaaa: Permission denied
result2 is ("getfacl: Removing leading '/' from absolute path names\n# owner: root\n# group: root\nuser::rwx\nuser:root:---\ngroup::r-x\ngroup:root:r--\ngroup:user1:r--\nmask::r-x\nother::---\n\n",)

======================================================================
FAIL: test_acl_1011 (__main__.TestFsrand2.test_acl_1011)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/juicefs/.github/scripts/hypo/acl_test.py", line 44, in test_acl_1011
    state.listdir(dir=v1, user='user1')
  File "/root/juicefs/.github/scripts/hypo/fs.py", line 176, in listdir
    user = st.sampled_from(SUDO_USERS))
         ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/dist-packages/hypothesis/stateful.py", line 736, in rule_wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/root/juicefs/.github/scripts/hypo/fs.py", line 176, in listdir
    user = st.sampled_from(SUDO_USERS))
         ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/dist-packages/hypothesis/stateful.py", line 857, in precondition_wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/root/juicefs/.github/scripts/hypo/fs.py", line 181, in listdir
    assert self.equal(result1, result2), f'\033[31mlistdir:\nresult1 is {result1}\nresult2 is {result2}\033[0m'
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: listdir:
result1 is ('aaaa',)
result2 is [Errno 13] Permission denied: '/tmp/jfs/fsrand/'

======================================================================
FAIL: test_acl_1022 (__main__.TestFsrand2.test_acl_1022)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/juicefs/.github/scripts/hypo/acl_test.py", line 63, in test_acl_1022
    state.set_acl(default=True, entry=v1, group='group4', group_perm={'x'}, logical=False, mask={'w', 'x'}, not_recalc_mask=False, other_perm=set(), physical=True, recalc_mask=False, recursive=True, set_mask=True, sudo_user='user1', user='user2', user_perm=set())
  File "/root/juicefs/.github/scripts/hypo/fs.py", line 417, in set_acl
    target=EntryWithACL,
         ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/dist-packages/hypothesis/stateful.py", line 736, in rule_wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/root/juicefs/.github/scripts/hypo/fs.py", line 417, in set_acl
    target=EntryWithACL,
         ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/dist-packages/hypothesis/stateful.py", line 857, in precondition_wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/root/juicefs/.github/scripts/hypo/fs.py", line 438, in set_acl
    assert self.equal(result1, result2), f'\033[31mset_acl:\nresult1 is {result1}\nresult2 is {result2}\033[0m'
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: set_acl:
result1 is
setfacl: /tmp/fsrand//lbca: Permission denied
setfacl: /tmp/fsrand/: Operation not permitted
result2 is
setfacl: /tmp/jfs/fsrand/: Operation not permitted

======================================================================
FAIL: test_acl_913 (__main__.TestFsrand2.test_acl_913)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/juicefs/.github/scripts/hypo/acl_test.py", line 12, in test_acl_913
    state.set_acl(default=False, entry=v1, group='root', group_perm=set(), logical=False, mask=set(), not_recalc_mask=False, other_perm=set(), physical=False, recalc_mask=False, recursive=True, set_mask=False, sudo_user='user1', user='root', user_perm=set())
  File "/root/juicefs/.github/scripts/hypo/fs.py", line 417, in set_acl
    target=EntryWithACL,
         ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/dist-packages/hypothesis/stateful.py", line 736, in rule_wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/root/juicefs/.github/scripts/hypo/fs.py", line 417, in set_acl
    target=EntryWithACL,
         ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/dist-packages/hypothesis/stateful.py", line 857, in precondition_wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/root/juicefs/.github/scripts/hypo/fs.py", line 438, in set_acl
    assert self.equal(result1, result2), f'\033[31mset_acl:\nresult1 is {result1}\nresult2 is {result2}\033[0m'
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: set_acl:
result1 is
setfacl: /tmp/fsrand//aaaa: Permission denied
setfacl: /tmp/fsrand/: Operation not permitted
result2 is
setfacl: /tmp/jfs/fsrand/: Operation not permitted

----------------------------------------------------------------------
Ran 12 tests in 3.411s

FAILED (failures=4)
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

--- a/pkg/vfs/vfs.go
+++ b/pkg/vfs/vfs.go
@@ -449,7 +449,12 @@ func (v *VFS) Readdir(ctx Context, ino Ino, size uint32, off int, fh uint64, plu
 		}
 		h.readAt = time.Now()
 		if h.dirHandler, err = v.Meta.NewDirHandler(ctx, ino, plus, initEntries); err != 0 {
-			return
+			if plus && err == syscall.EACCES {
+				h.dirHandler, err = v.Meta.NewDirHandler(ctx, ino, false, initEntries)
+			}
+			if err != 0 {
+				return
+			}
 		}
 	}
 	if entries, err = h.dirHandler.List(ctx, off); err != 0 {

